homeassistant/util/unit_system.py

```diff
@@ -100,12 +100,37 @@ def temperature(self: object, temperature: float, from_unit: str) -> float:
                                         from_unit, self.temperature_unit)
 
     def length(self: object, length: float, from_unit: str) -> float:
-        """Convert the given length to this unit system."""
+        """Convert the given length to this unit system return a float."""
+        converted = self.length_and_unit(length, from_unit, False)
+        return converted["value"]
+
+    def length_and_unit(
+            self: object, length: float, from_unit: str,
+            auto_range: bool) -> dict:
```

Comment:
The autorange boolean makes no sense, this method is about "smart" conversion and this boolean turns it off. Please remove this and just always do it.

Reply:
This goes hand in hand with the above comment